app/assets/javascripts/Measure.js

```diff
@@ -0,0 +1,70 @@
+const mongoose = require('mongoose');
+const Code = require('./basetypes/Code');
+const Interval = require('./basetypes/Interval');
+const Quantity = require('./basetypes/Quantity');
+
+const [Integer, Array, String, Boolean, Mixed, ObjectId, Date] = [
+  mongoose.Schema.Types.Integer,
+  mongoose.Schema.Types.Array,
+  mongoose.Schema.Types.String,
+  mongoose.Schema.Types.Boolean,
+  mongoose.Schema.Types.Mixed,
+  mongoose.Schema.Types.ObjectId,
+  mongoose.Schema.Types.Date,
+];
+
+const MeasureSchema = mongoose.Schema(
+  {
+    id: String,
+    measure_id: String,
+    hqmf_id: String,
+    hqmf_set_id: String,
+    hqmf_version_number: Integer,
+    cms_id: String,
+    title: String,
+    description: String,
+    type: String,
+    category: { type: String, default: 'Uncategorized' },
+
+    episode_of_care: Boolean,
+    continuous_constiable: Boolean,
+    episode_ids: [], // not sure if this is necessary
+
+    needs_finalize: Boolean, // Bonnie-specific?
+
+    published: Boolean,
+    publish_date: Date,
+    version: Integer,
+
+    elm_annotations: Mixed,
+
+    cql: [String],
+    elm: [Mixed],
+    main_cql_library: String,
+    cql_statement_dependencies: Mixed,
+
+    population_criteria: Mixed,
+    data_criteria: Mixed,
+    source_data_criteria: Mixed,
+    measure_period: Interval,
+    measure_attributes: [],
+    populations: [Mixed],
+    populations_cql_map: Mixed,
+    observations: [Mixed],
+
+    value_sets: [{ type: ObjectId, ref: 'ValueSet' }],
+
+    complexity: Mixed, // Bonnie-specific?
+
+    user: { type: ObjectId, ref: 'User', index: true }, // Bonnie-specific
+    bundle: { type: ObjectId, ref: 'Bundle' }, // Cypress-specific, but used in Bonnie-bundler as well
+    package: { type: ObjectId, ref: 'MeasurePackage' }, // Bonnie specific
+  },
+  // Options
+  {
+    timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }, // These are the Mongoid conventions for timestamps
+  },
+);
+
+module.exports.MeasureSchema = MeasureSchema;
+module.exports.Measure = mongoose.model('measure', MeasureSchema);
```

Comment:
The models look good, but you're going to want to generate them now using the the generate_models.rb file, probably the same methodology as how we generate the patient model. Especially since regenerating the files requires deleting all of the ruby models each time before executing the regeneration (which would delete the measure model in ruby too).

Reply:
That sounds like a good idea @ljades, but I'm not certain how exactly that would look. Patient is actually a datatype in QDM, so the generator just extends it (using the patient_extention template). Since Measure isn't, we'd have to do it a bit differently. Perhaps put the entire Measure models in their corresponding .erb templates, and then render and save those as new files in the post processing phase.

Reply:
That could work. Another option is placing it in a subfolder, as we do with the basetypes, which basically says, "Don't delete pls."

Reply:
Yeah; I'm not a huge fan of the template-based solution, if they're just being spit out without any changes. The subfolder solution might work.

Reply:
Sounds good. I don't think it should go into `basetypes` though. Maybe have another subdirectory called `tacoma` or something that contains models specific for cypress/bonnie?

Reply:
@holmesie / @ljades I moved the measure/VS related models into `tacoma` to get them out of the `qdm` directory so they won't get "post-processed"